spec/steps/provision.fmf

```diff
@@ -115,3 +114,64 @@ example: |
             image: 1MT-Fedora-32
             flavor: m1.large
     implemented: /tmt/steps/provision/minute.py
+
+/environment:
+    summary: Environment specification
+
+    description: |
+        As part of the provision step it is possible to specify
+        additional requirements for the testing environment.
+        Individual requirements are provided as a simple ``key:
+        value`` pairs, for example the minimum amount of
+        ``memory`` and ``disk`` space, or the related information
+        is grouped under a common parent, for example ``cores`` or
+        ``model`` under the ``cpu`` key.
+
+        When multiple environment requirements are provided the
+        provision implementation should attempt to satisfy all of
+        them. It is also possible to write this explicitly using
+        the ``and`` operator containing a list of dictionaries
+        with individual requirements. When the ``or`` operator is
+        used, any of the alternatives provided in the list should
+        be sufficient.
+
+        Regular expressions can be used for selected fields such
+        as the ``model_name``. Please note, that the full extent
+        of regular expressions might not be supported across all
+        provision implementations. The ``.*`` notation, however,
+        should be supported everywhere.
+
+    example: |
+        # Simple use cases simple
+        memory: 8 GB
+        disk: 500 GB
+
+        # Processor-related stuff grouped together
+        cpu:
+            processors: 2
+            cores: 16
+            model: 37
+
+        # Optional operators at the start of the value
```

Comment:
Same reply as to @thrix: The original idea was to kick-off the spec with just a bunch of initial examples so that we have at least something documented soon. But if we agree it would be better to deep dive already in this pull request... we can change that and spend some more time with it.